spec/models/credentials_spec.rb

```diff
@@ -57,10 +57,82 @@
           expect(credentials.valid_password?(nil)).to be_falsey
         end
       end
+      it 'allows passwords with 128 characters long' do
+        # 128 characters long (Requires: At least 12 characters and maximum of 128)
+        credentials.password = "My Password134567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678"
+        expect { credentials.save }.not_to raise_error
+      end
+      it 'allows passwords with all acceptable special characters' do
+        # with all acceptable special characters
+        credentials.password = "New-Password1 !\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~]"
+        expect { credentials.save }.not_to raise_error
+      end
+      it 'allows passwords with mixed languages (Russian and French)' do
+        # with mixed languages (Russian and French)
+        credentials.password = "New!Password1БбИиàâæ"
+        expect { credentials.save }.not_to raise_error
+      end
       it 'disallows passwords with newlines' do
```

Comment:
not corrent. we don't check that `\n` is not allowed since this use case `foo\nbar` is not answering required complexity.